### ISSUE TITLE
Regression: Issues with Layer selection in GeoFence

### ISSUE THREAD

**Issue by offtherailz** [geosolutions-it/MapStore2-C040#22]
From the UI of geofence, selecting a workspace you can not select the layer (from the workspace).
Probably due to some misconfiguration on layers.
 only production issues, it works in test: 

![image](https://cloud.githubusercontent.com/assets/1279510/23254075/112d4574-f9b7-11e6-9de7-8dd14d3dff4e.png)

**Comment by nmco** [comment]
The log of this issue on GeoServer side is:

```
2017-02-28 23:34:03,599 WARN [geoserver.monitor] - Post process task failed
java.lang.IllegalStateException: No thread-bound request found: Are you referring to request attributes outside of an actual web request, or processing a request outside of the originally receiving thread? If you are actually operating within a web request and still receive this message, your code is probably running outside of DispatcherServlet/DispatcherPortlet: In this case, use RequestContextListener or RequestContextFilter to expose the current request.
	at org.springframework.web.context.request.RequestContextHolder.currentRequestAttributes(RequestContextHolder.java:131)
	at org.geoserver.geofence.GeofenceAccessManager.retrieveCallerIpAddress(GeofenceAccessManager.java:253)
	at org.geoserver.geofence.GeofenceAccessManager.getAccessLimits(GeofenceAccessManager.java:367)
	at org.geoserver.geofence.GeofenceAccessManager.getAccessLimits(GeofenceAccessManager.java:293)
	at org.geoserver.security.ResourceAccessManagerWrapper.getAccessLimits(ResourceAccessManagerWrapper.java:214)
	at org.geoserver.security.CatalogFilterAccessManager.getAccessLimits(CatalogFilterAccessManager.java:56)
	at org.geoserver.security.SecureCatalogImpl.buildWrapperPolicy(SecureCatalogImpl.java:806)
	at org.geoserver.security.SecureCatalogImpl.buildWrapperPolicy(SecureCatalogImpl.java:770)
	at org.geoserver.security.SecureCatalogImpl.checkAccess(SecureCatalogImpl.java:594)
	at org.geoserver.security.SecureCatalogImpl.getLayerByName(SecureCatalogImpl.java:295)
	at org.geoserver.catalog.impl.AbstractFilteredCatalog.getLayerByName(AbstractFilteredCatalog.java:225)
	at org.geoserver.catalog.impl.AbstractCatalogDecorator.getLayerByName(AbstractCatalogDecorator.java:384)
	at org.geoserver.catalog.impl.LocalWorkspaceCatalog.getLayerByName(LocalWorkspaceCatalog.java:92)
	at org.geoserver.monitor.LayerNameNormalizer.run(LayerNameNormalizer.java:43)
	at org.geoserver.monitor.MonitorFilter$PostProcessTask.run(MonitorFilter.java:246)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

The request that produce this issue (with passwords and user names censured):

curl --request POST \
  --url http://mappe.comune.genova.it/geofence/geofence/WorkspacesManagerRemoteService \
  --header 'authorization: Basic YWRtaW46IV9jb211bmUqR0VfIQ==' \
  --header 'content-type: text/x-gwt-rpc; charset=utf-8' \
  --header 'x-gwt-permutation: C6CCC01B6480E54A5A2CA8113BD3A6AB' \
  --cookie JSESSIONID=E3A49DC54B60034892210563A6B781FE \
  --data '7|0|23|http://mappe.comune.genova.it/geofence/geofence/|5C83DAC3FE4B1681CBB9FB7537AA1445|org.geoserver.geofence.gui.client.service.WorkspacesManagerRemoteService|getLayers|I|java.lang.String/2004016611|org.geoserver.geofence.gui.client.model.GSInstanceModel/1063805272|http://mappe.comune.genova.it/geoserver-master/|java.sql.Timestamp/3040052672|default-gs|<PASSWORD>|geofence/resources/images/instance.jpg|admin|java.util.ArrayList/4159755760|com.extjs.gxt.ui.client.core.FastMap/3488076414|com.extjs.gxt.ui.client.data.RpcMap/3441186752|baseUrl|dateCreation|description|name|path|SITGEO|*|1|2|3|4|6|5|5|6|7|6|6|0|0|8|7|8|9|U1HoFxI|288000000|10|F|10|11|12|13|14|0|15|0|1|16|5|17|6|8|18|-2|19|6|10|20|-7|21|6|12|22|23|'

Long story short, this is actually a simple configuration issue the production GeoServer masters and slaves nodes are using the QA environment GeoFence:
```
http://vm-linuxgeofetest.comune.genova.it:8083/geofence/remoting/RuleReader
```
and production GeoFence is pointing to QA test environment GeoServer:
```
http://vm-linuxgeofe.comune.genova.it:8082/geoserver-master
```

The rules on both environments are pretty much the same.

**Comment by offtherailz** [comment]
This issue will be solved updating GeoFence in production.
